src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -36,33 +56,72 @@ public FindCommand parse(String args) throws ParseException {
+            name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+            String[] nameList = new String[] {name.toString()};
+            predicate = predicate.and(new NameContainsKeywordsPredicate(Arrays.asList(nameList)));
+            builder.setName(name);
+        }
+        if (argMultimap.getValue(PREFIX_GENDER).isPresent()) {
+            gender = ParserUtil.parseGender(argMultimap.getValue(PREFIX_GENDER).get());
+            Gender[] genderList = new Gender[] {gender};
+            predicate = predicate.and(new GenderContainsGenderPredicate(Arrays.asList(genderList)));
+            builder.setGender(gender);
+        }
+        if (argMultimap.getValue(PREFIX_QUALIFICATION).isPresent()) {
+            // Guard clause in case user searches student using qualification
+            if (personType == PersonType.STUDENT) {
+                throw new ParseException(MESSAGE_INVALID_INPUT_STUDENT_WITH_QUALIFICATION);
+            }
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME)
-                || argMultimap.getPreamble().isEmpty()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            qualification =
+                    ParserUtil.parseQualification(argMultimap.getValue(PREFIX_QUALIFICATION).get());
+            Qualification[] qualificationList = new Qualification[] {qualification};
+            predicate =
+                    predicate.and(new QualificationContainsQualificationPredicate(Arrays.asList(qualificationList)));
+            builder.setQualification(qualification);
+        }
+        if (parseTags(argMultimap.getAllValues(PREFIX_TAG)).isPresent()) {
+            tags = parseTags(argMultimap.getAllValues(PREFIX_TAG)).get();
+            predicate = predicate.and(new TagsContainTagPredicate(tags));
+            builder.setTags(tags);
         }
 
-        return ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        return builder.setPredicate(predicate).build();
     }
 
     /**
-     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
-     * {@code ArgumentMultimap}.
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>} if {@code tags} is non-empty.
+     * If {@code tags} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Tag>} containing zero tags.
      */
-    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
-        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    private Optional<List<Tag>> parseTags(Collection<String> tags) throws ParseException {
+        assert tags != null;
+
+        if (tags.isEmpty()) {
+            return Optional.empty();
+        }
+
+        Collection<String> tagSet = tags.size() == 1 && tags.contains("") ? Collections.emptySet() : tags;
```

Comment:
This might be a slightly complicated expression? Maybe split it up or write it in a more readable format?